apt-hook/hook.cc

```diff
@@ -201,69 +310,67 @@ int main(int argc, char *argv[])
-      {
-         ioprintf(std::cout,
-                  ngettext("%d of the updates is from UA Infra: ESM.",
-                           "%d of the updates are from UA Infra: ESM.",
-                           res.enabled_esms_i),
-                  res.enabled_esms_i);
-         ioprintf(std::cout, "\n");
-      }
-      if (res.enabled_esms_a > 0)
-      {
-         ioprintf(std::cout,
-                  ngettext("%d of the updates is from UA Apps: ESM.",
-                           "%d of the updates are from UA Apps: ESM.",
-                           res.enabled_esms_a),
-                  res.enabled_esms_a);
-         ioprintf(std::cout, "\n");
+   // Compute all strings necessary to fill in templates
+   std::string space_separated_esm_i_packages = "";
+   if (res.esm_i_packages.size() > 0) {
+      for (uint i = 0; i < res.esm_i_packages.size() - 1; i++) {
+         space_separated_esm_i_packages.append(res.esm_i_packages[i]);
+         space_separated_esm_i_packages.append(" ");
       }
+      space_separated_esm_i_packages.append(res.esm_i_packages[res.esm_i_packages.size() - 1]);
    }
-
-   if (res.disabled_esms_i > 0 || res.disabled_esms_a > 0)
-   {
-      if (command_used != "update")
-         std::cout << std::endl;
-      if (res.disabled_esms_i > 0)
-      {
-         ioprintf(std::cout,
-                  ngettext("%d additional update is available with UA Infra: ESM.",
-                           "%d additional updates are available with UA Infra: ESM.",
-                           res.disabled_esms_i),
-                  res.disabled_esms_i);
-         ioprintf(std::cout, "\n");
+   std::string space_separated_esm_a_packages = "";
+   if (res.esm_a_packages.size() > 0) {
+      for (uint i = 0; i < res.esm_a_packages.size() - 1; i++) {
+         space_separated_esm_a_packages.append(res.esm_a_packages[i]);
+         space_separated_esm_a_packages.append(" ");
       }
-      if (res.disabled_esms_a > 0)
-      {
-         ioprintf(std::cout,
-                  ngettext("%d additional update is available with UA Apps: ESM.",
-                           "%d additional updates are available with UA Apps: ESM.",
-                           res.disabled_esms_a),
-                  res.disabled_esms_a);
-         ioprintf(std::cout, "\n");
+      space_separated_esm_a_packages.append(res.esm_a_packages[res.esm_a_packages.size() - 1]);
+   }
+   std::string esm_i_packages_count = std::to_string(res.esm_i_packages.size());
+   std::string esm_a_packages_count = std::to_string(res.esm_a_packages.size());
+
+   process_all_templates(
+      esm_a_packages_count,
+      space_separated_esm_a_packages,
+      esm_i_packages_count,
+      space_separated_esm_i_packages
+   );
+
+   // Execute specified subcommand
+   if (subcommand == PreInvoke) {
+      if (command_used == "upgrade" || command_used == "dist-upgrade") {
+         output_file_if_present(CONTRACT_EXPIRY_STATUS_APT_MESSAGE_STATIC_PATH);
```

Comment:
I had envisioned these two templates as separate sub-messages
any objection to the following?
```suggestion
         output_file_if_present(CONTRACT_EXPIRY_STATUS_MESSAGE_STATIC_PATH);         
         output_file_if_present(CONTRACT_EXPIRY_STATUS_APT_MESSAGE_STATIC_PATH);
```

this way I think we can use a common expiry message/banner for both motd and APT and extend that info for apt providing supplementary messaging.

Reply:
Ahhhh - totally that makes sense. I'll make the change and adjust the integration test

Reply:
FFR: We double checked - motd and apt don't consistently share a header, so we won't apply this change

Reply:
Ack, we'll close this a Chad's mistaken.